src/core/tracing/Window.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Override some bits of foam.core.Window to support more accurate tracing.
+foam.CLASS({
+  package: 'foam.core.tracing',
+  name: 'Window',
+  extends: 'foam.core.Window',
```

Comment:
Why does this need to extend Window?
When I create a new Window, how will I know to add this context as well?